packages/ui-text-input/src/TextInput/index.tsx

```diff
@@ -97,6 +97,10 @@ class TextInput extends Component<TextInputProps, TextInputState> {
   }
   componentDidMount() {
     this.props.makeStyles?.(this.makeStyleProps())
+    document
+      .getElementById(this.id)
+      //@ts-expect-error TODO
+      ?.addEventListener('focus', this.handleFocus)
```

Comment:
Nice catch :) I would recommend using our own addEventListener util, and add a comment above it with the issue link and an explanation for the workaround. Also, please always remove listeners on unmount in order to prevent memory leaks. I found a similar workaround in RangeInput, you can use that as a guide :)